devicetypes/timevalve-gaslock-t-08/timevalve-smart.src/timevalve-smart.groovy

```diff
@@ -0,0 +1,281 @@
+                state "10", label:'10Min', action: "setTimeout10", icon:"st.Office.office13"
+            }
+            standardTile("setTimeout20", "device.remaining", inactiveLabel: false, decoration: "flat") {
+                state "default", label:'20Min', action: "setTimeout20", icon:"st.Health & Wellness.health7", defaultState: true
+                state "20", label:'20Min', action: "setTimeout20", icon:"st.Office.office13"
+            }
+            standardTile("setTimeout30", "device.remaining", inactiveLabel: false, decoration: "flat") {
+                state "default", label:'30Min', action: "setTimeout30", icon:"st.Health & Wellness.health7", defaultState: true
+                state "30", label:'30Min', action: "setTimeout30", icon:"st.Office.office13"
+            }
+            standardTile("setTimeout40", "device.remaining", inactiveLabel: false, decoration: "flat") {
+                state "default", label:'40Min', action: "setTimeout40", icon:"st.Health & Wellness.health7", defaultState: true
+                state "40", label:'40Min', action: "setTimeout40", icon:"st.Office.office13"
+            }
+
+            valueTile("batteryTile", "device.battery", width: 2, height: 2, inactiveLabel: false, decoration: "flat") {
+                state "battery", label:'${currentValue}% battery', unit:""
+            }
+
+            main (["statusTile"])
+//            details (["statusTile", "remainingSliderTile", "setRemaining", "setTimeout10", "setTimeout20", "batteryTile", "refreshTile", "setTimeout30", "setTimeout40"])
+//            details (["statusTile", "batteryTile", "setRemaining", "refreshTile"])
+            details (["statusTile", "batteryTile", "refreshTile"])
+        }
+}
+
+def parse(description) {
+//	log.debug "parse - " + description
+	def result = null
+	if (description.startsWith("Err 106")) {
+		state.sec = 0
+		result = createEvent(descriptionText: description, isStateChange: true)
+	} else if (description != "updated") {
+		def cmd = zwave.parse(description, [0x20: 1, 0x25: 1, 0x70: 1, 0x71: 1, 0x98: 1])
+		if (cmd) {
+        	log.debug "parsed cmd = " + cmd
+			result = zwaveEvent(cmd)
+			//log.debug("'$description' parsed to $result")
+		} else {
+			log.debug("Couldn't zwave.parse '$description'")
+		}
+	}
+	return result
+}
+
+// 복호화 후 zwaveEvent() 호출
+def zwaveEvent(physicalgraph.zwave.commands.securityv1.SecurityMessageEncapsulation cmd) {
+	//log.debug "SecurityMessageEncapsulation - " + cmd
+	def encapsulatedCommand = cmd.encapsulatedCommand([0x20: 1, 0x25: 1, 0x70: 1, 0x71: 1, 0x98: 1])
+	if (encapsulatedCommand) {
+		state.sec = 1
+        log.debug "encapsulatedCommand = " + encapsulatedCommand
+        zwaveEvent(encapsulatedCommand)
+	}
+}
+
+def zwaveEvent(physicalgraph.zwave.commands.switchbinaryv1.SwitchBinaryReport cmd) {
+	//log.debug "switch status - " + cmd.value
+    createEvent(name:"contact", value: cmd.value ? "open" : "closed")
```

Comment:
valve is capability name.
createEvent 'name' need attribute name. right?
now work fine.